### PR TITLE
Refine mea authorization

### PR DESCRIPTION
**Pull request by Ashish0410verma**

### CODE REVIEW

File: src/Kmd.Momentum.Mea.Api/appsettings.Internal.json [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
       "KommuneId": "185",
       "KommuneUrl": "https://kmd-rct-momentum-159-api.azurewebsites.net/api/",
-      "Scopes": {
-        "ScopeForCitizenApi": "access",
-        "ScopeForCaseworkerApi": "",
-        "ScopeForJournalApi": ""
-      }
+      "KommuneClientId": "4a7a4c73-f203-435e-b5c2-3cbba12f0285",
+      "KommuneAccessIdentifier": "kmd-internal-momentum-185-mea-secret",
+      "KommuneResource": "74b4f45c-4e9b-4be1-98f1-ea876d9edd11"
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/appsettings.Internal.json:20]: object id for this is different

File: src/Kmd.Momentum.Mea.Api/appsettings.Phoenix.json [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
       "KommuneId": "185",
       "KommuneUrl": "https://kmd-rct-momentum-159-api.azurewebsites.net/api/",
-      "Scopes": {
-        "ScopeForCitizenApi": "access",
-        "ScopeForCaseworkerApi": "",
-        "ScopeForJournalApi": ""
-      }
+      "KommuneClientId": "4a7a4c73-f203-435e-b5c2-3cbba12f0285",
+      "KommuneAccessIdentifier": "kmd-phoenix-momentum-159-mea-secret",
+      "KommuneResource": "74b4f45c-4e9b-4be1-98f1-ea876d9edd11"
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/appsettings.Phoenix.json:20]: again..

File: src/Kmd.Momentum.Mea.Api/appsettings.Internal.json (hunk 1 above)
Comment (Ashish0410verma) [reply src/Kmd.Momentum.Mea.Api/appsettings.Internal.json:20]: Will put later as for testing we are gonna use 159

File: src/Kmd.Momentum.Mea.Api/appsettings.Phoenix.json (hunk 2 above)
Comment (Ashish0410verma) [reply src/Kmd.Momentum.Mea.Api/appsettings.Phoenix.json:20]: replied above

File: src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs [hunk 3]
```diff
@@ -101,33 +118,84 @@ public MeaClient(IConfiguration config, HttpClient httpClient, IHttpContextAcces
         }
 
         private async Task<ResultOrHttpError<HttpResponseMessage, string>> ReturnAuthorizationTokenAsync()
+        {
+            var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Development";
+
+            if (environment == "Build")
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs:124]: inside if and else lot of code is repeated. SO basically if should only contain how we are defining the secret. rest all the code should be out of ifelse.

Comment (Ashish0410verma) [reply src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs:125]: fixed

File: src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs [hunk 4]
```diff
@@ -17,15 +20,23 @@ public class MeaClient : IMeaClient
     {
         private static HttpClient _httpClient;
         private readonly IConfiguration _config;
+        private readonly IMeaSecretStore _meaSecretStore;
         private readonly string _correlationId;
+        private readonly string _tenant;
+        private readonly MeaAuthorization _mcaConfig;
+        private string _mcaClientSecret;
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs:27]: we dont need this global variable

File: src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs [hunk 5]
```diff
@@ -101,33 +119,74 @@ public MeaClient(IConfiguration config, HttpClient httpClient, IHttpContextAcces
         }
 
         private async Task<ResultOrHttpError<HttpResponseMessage, string>> ReturnAuthorizationTokenAsync()
+        {
+            var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Development";
+
+            if (environment == "Build")
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs:125]: put this entire logic in a private method

File: src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs (hunk 4 above)
Comment (Ashish0410verma) [reply src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs]: fixed

File: src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs (hunk 5 above)
Comment (Ashish0410verma) [reply src/Kmd.Momentum.Mea.Common/MeaHttpClient/MeaClient.cs]: fixed

**Pull request merged**